src/PIL/ImageGrab.py

```diff
@@ -61,7 +61,9 @@ def grab(bbox=None, include_layered_windows=False, all_screens=False, xdisplay=N
                 left, top, right, bottom = bbox
                 im = im.crop((left - x0, top - y0, right - x0, bottom - y0))
             return im
-        elif shutil.which("gnome-screenshot"):
+        elif not (
+            Image.core.HAVE_XCB and os.environ.get("XDG_SESSION_TYPE") == "x11"
```

Comment:
What if `XDG_SESSION_TYPE` is not set? Or alternatively, if the X11 connection is not configured properly (e.g. `DISPLAY` is not set)? The following is set on my Ubuntu 20.04 (in WSL), with the `DISPLAY` value being valid only when I do not forget to run my X server. I imagine the `DISPLAY` value could be invalid for other reasons as well.

```bash
$ set | grep DISPLAY
DISPLAY=:0
$ set | grep XDG
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
    local -a dirs=(${BASH_COMPLETION_USER_DIR:-${XDG_DATA_HOME:-$HOME/.local/share}/bash-completion}/completions);
    for dir in ${XDG_DATA_DIRS:-/usr/local/share:/usr/share};
$
```

I think a better option is to use a try-except block instead to prefer X11 over gnome-screenshot: #7143